e2e/test/scenarios/visualizations/table.cy.spec.js

```diff
@@ -269,6 +270,46 @@ describe("scenarios > visualizations > table", () => {
     cy.wait(100);
     popover().should("not.exist");
   });
+
+  it("popover should not be horizontally scrollable (metabase#31339)", () => {
+    openPeopleTable();
+    headerCells().filter(":contains('Password')").click();
+
+    popover().within(() => {
+      cy.findByText("Filter by this column").click();
+      cy.findByPlaceholderText("Search by Password").type("e").blur();
+      cy.wait("@findSuggestions");
+    });
+
+    popover().then(popoverElement => {
+      expect(
+        popoverElement[0].clientHeight,
+        "horizontal scrollbar is not shown",
+      ).to.eq(popoverElement[0].offsetHeight);
+    });
+  });
+
+  it("default picker container should not be horizontally scrollable", () => {
+    openPeopleTable();
+    headerCells().filter(":contains('Password')").click();
+
+    popover().within(() => {
+      cy.findByText("Filter by this column").click();
+
+      const input = cy.findByPlaceholderText("Search by Password");
+      input.type("e").blur();
+      cy.wait("@findSuggestions");
+      input.type("f");
+      cy.wait("@findSuggestions");
+
+      cy.findByTestId("default-picker-container").then(containerElement => {
```

Comment:
Would it be possible to use `be.visible` or something like that? I believe it checks for overlapping

Reply:
Here were are comparing `offsetHeight` and `clientHeight` of **the same element**, to detect whether horizontal scrollbar is visible. There is no overlapping in this case AFAIK.

`be.visible` does not help, I didn't find anything else. Perhaps we need to extract this helper. I'll remember it next time I need it.